src/main/views/courts/tabs/addressesContent.njk

```diff
@@ -0,0 +1,178 @@
+  name: "primary[address_lines_cy]",
+  label: {
+    text: "Address (Welsh)"
+  },
+  classes: "govuk-input--width-30",
+  value: addresses.primary.address_lines_cy
+}) }}
+
+{{ govukInput({
+  id: "primaryAddressTown",
+  name: "primary[town]",
+  label: { text: "Town" },
+  classes: "govuk-input--width-30",
+  value: addresses.primary.town,
+  errorMessage: { text: 'Town is required' } if addresses.primary.town === ''
+}) }}
+
+{{ govukInput({
+  id: "primaryAddressTownWelsh",
+  name: "primary[town_cy]",
+  label: { text: "Town (Welsh)" },
+  classes: "govuk-input--width-30",
+  value: addresses.primary.town_cy
+}) }}
+
+{% if addresses.primary.postcode == '' %}
+  {% set primaryPostcodeError = { text: 'Postcode is required' } %}
+{% elif primaryPostcodeInvalid %}
+  {% set primaryPostcodeError = { text: 'Postcode is invalid' } %}
+{% endif %}
+
+{{ govukInput({
+  id: "primaryAddressPostcode",
+  name: "primary[postcode]",
+  label: { text: "Postcode" },
+  classes: "govuk-input--width-30",
+  value: addresses.primary.postcode,
+  errorMessage: primaryPostcodeError
+}) }}
+
+<br/>
+
+<h3 class="govuk-heading-m">Secondary Address</h3>
+
+{{ govukSelect({
+  id: "secondaryAddressType",
+  name: "secondary[type_id]",
+  label: { text: "Address Type" },
+  items: addressTypesSecondary  | selectFilter(addresses.secondary.type_id)
+}) }}
+
+{% if (addresses.secondary.address_lines) or (addresses.secondary.address_lines_cy) or (addresses.secondary.town) or (addresses.secondary.town_cy) or (addresses.secondary.postcode) %}
+  {% if addresses.secondary.address_lines == '' %}
+    {% set secondaryAddressError = { text: 'Address is required' } %}
+  {% endif %}
+  {% if addresses.secondary.town == '' %}
+    {% set secondaryTownError = { text: 'Town is required' } %}
+  {% endif %}
+{% endif %}
+
+{{ govukTextarea({
+  id: "secondaryAddressLines",
+  name: "secondary[address_lines]",
+  label: { text: "Address" },
+  classes: "govuk-input--width-30",
+  value: addresses.secondary.address_lines,
+  errorMessage: secondaryAddressError
+}) }}
+
+{{ govukTextarea({
+  id: "secondaryAddressLinesWelsh",
+  name: "secondary[address_lines_cy]",
+  label: { text: "Address (Welsh)" },
+  classes: "govuk-input--width-30",
+  value: addresses.secondary.address_lines_cy
+}) }}
+
+{{ govukInput({
+  id: "secondaryAddressTown",
+  name: "secondary[town]",
+  label: { text: "Town" },
+  classes: "govuk-input--width-30",
+  value: addresses.secondary.town,
+  errorMessage: secondaryTownError
+}) }}
+
+{{ govukInput({
+  id: "secondaryAddressTownWelsh",
+  name: "secondary[town_cy]",
+  label: { text: "Town (Welsh)" },
+  classes: "govuk-input--width-30",
+  value: addresses.secondary.town_cy
+}) }}
+
+{% if addresses.secondary.postcode == '' %}
+  {% if addresses.secondary.address_lines or addresses.secondary.address_lines_cy or addresses.secondary.town or addresses.secondary.town_cy %}
+    {% set secondaryPostcodeError = { text: 'Postcode is required' } %}
```

Comment:
A small one: 'Postcode is required' and 'Postcode is invalid' are both used more than once, is it better for us to set them as variables and use them in here? Possibly the same for the others too if you think it's worth it

Reply:
Good catch. I've updated the view.